### PR TITLE
Add 'easy to install' base and micro profiles, with multi ns

### PR DESCRIPTION
**Pull request by costinm**
Same approach as the demo - pre-generate images with the recommended values, user can 
install with a single command.

Both should work alongside istio-1.1

### CODE REVIEW

**Comment by costinm** [comment]
Example:
  kubectl apply -k github.com/costinm/istio-installer/kustomize/base/ ( will be istio/installer after merge )

File: kustomize/micro/allocation.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  template:
+    spec:
+      containers:
+        - name: istio-proxy
+          resources:
+            limits:
+              memory: 1250Mi
+            requests:
+              memory: 1250Mi
```
Comment (howardjohn) [inline kustomize/micro/allocation.yaml]: Can we get a readme or something on the profiles? I thought micro meant small install (in terms of resources) but based on this I guess its micro in terms of number of components?

Comment (costinm) [reply kustomize/micro/allocation.yaml]: Yes, it is on number of components. 
The allocation file is mostly to test that it works and provide an example - I'll adjust this.

**Comment by costinm** [comment]
On generated comment: I agree, there are few options long term ( different branch, hosted ).
I don't remember if kustomize requires a git repo or any URL would work.

But short term - I think there are benefits in having the diff on generated files, and it seems to
work. The files have comments that they were generated. 

Also: if at some point we simplify the helm templates for a component to the point of having 
plain yaml files without templating  - we can use them as the source for kustomize. See the CRD directory for example.

File: kustomize/istio-ingress/istio-ingress.yaml [hunk 2]
```diff
@@ -96,18 +109,17 @@ spec:
...
-          - --log_output_level
-          - 'info'
-          - --proxyLogLevel
-          - 'info'
+          - --proxyLogLevel=warning
+          - --proxyComponentLogLevel=misc:error
+          - --log_output_level=default:info
           - --drainDuration
-          - '45s'
+          - '45s' #drainDuration
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply kustomize/istio-ingress/istio-ingress.yaml]: this is generated. but the comment is used in some test for some reason.. not sure why

[the remark this answers is not in the document]
Comment (costinm) [reply kustomize/istio-ingress/istio-ingress.yaml]: Either way: we should make this customizable.  Kustomize doesn't work well with CLI, so likely needs to be in values.yaml.  I added similar options in other places for connect timeout I think.

But can be done in separate PR.

File: kustomize/micro/allocation.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: ingressgateway
+spec:
+  template:
+    spec:
+      containers:
+        - name: istio-proxy
+          resources:
+            limits:
+              memory: 1250Mi
```
[the remark this answers is not in the document]
Comment (costinm) [reply kustomize/micro/allocation.yaml]: The intention is to provide an example on how kustomize can specify allocations. I'll make sure the values here are the same we have in default install.

File: kustomize/micro/istio-ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - s390x      
+
+---
+# Source: istio-ingress/templates/gateway.yaml
+
+# Main Gatewaay. Apps must bind to NAMESPACE/ingressgateway
```
[the remark this answers is not in the document]
Comment (costinm) [reply kustomize/micro/istio-ingress.yaml]: fixed in the source

File: kustomize/micro/istio-ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+          - --proxyAdminPort
+          - "15000"
+          - --statusPort
+          - "15020"
+          - --controlPlaneAuthPolicy
+          - NONE
+          - --discoveryAddress
+          - istio-pilot.istio-micro:15010
+          readinessProbe:
+            failureThreshold: 30
```
[the remark this answers is not in the document]
Comment (costinm) [reply kustomize/micro/istio-ingress.yaml]: That's what we have in the template - probably not the right value but something to fine tune in separate PR.

**Comment by costinm** [comment]
PTAL - demo can now also be installed as an in-place upgrade over 1.1 ( TODO: add a test for this - but the P0 is upgrade tests for the main flow, not demo)

File: docs/labels.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+= Upgrade and deployment labels
```
Comment (howardjohn) [inline docs/labels.md]: nit: I don't think the `= title` works in github

File: istio-control/istio-autoinject/templates/deployment.yaml [hunk 7]
```diff
@@ -19,9 +19,13 @@ spec:
   template:
     metadata:
       labels:
-        app: sidecar-injector
-        release: {{ .Release.Name }}
+        app: sidecarInjectorWebhook
```
Comment (howardjohn) [inline istio-control/istio-autoinject/templates/deployment.yaml]: good catch

**Pull request merged**